File: leetcode/121. Best Time to Buy and Sell Stock/BestTimeStock.cpp
class Solution
{
public:
    int maxProfit(vector<int> &prices)
    {
        int minPrice = prices[0], maxProfit = 0;
        for (const int price : prices)
        {
            int profit = price - minPrice;
            maxProfit = max(maxProfit, profit);
            minPrice = min(price, minPrice);
        }
        return maxProfit;
    }
};

File: leetcode/121. Best Time to Buy and Sell Stock/BestTimeStock.go
func maxProfit(prices []int) int {
	minPrice, maxProfit := prices[0], 0
	for _, price := range prices {
		profit := price - minPrice
		maxProfit = max(maxProfit, profit)
		minPrice = min(minPrice, price)
	}
	return maxProfit
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}